ReactSkia/views/common/RSkImageCacheManager.cpp

```diff
@@ -11,130 +11,115 @@
 #include "ReactSkia/RSkSurfaceWindow.h"
 #include "ReactSkia/views/common/RSkImageCacheManager.h"
 #include "ReactSkia/utils/RnsLog.h"
-
+#include "rns_shell/common/WindowContext.h"
 using namespace std;
 
 #define SKIA_CPU_IMAGE_CACHE_HWM_LIMIT  SKIA_CPU_IMAGE_CACHE_LIMIT *.95 //95% as High Water mark level
 #define SKIA_GPU_IMAGE_CACHE_HWM_LIMIT  SKIA_CPU_IMAGE_CACHE_LIMIT *.95 //95% as High Water mark level
 #define EVICT_COUNT  2 // Max No. of entries to be evicted in single run
 
+#ifdef RNS_SHELL_HAS_GPU_SUPPORT
+std::mutex RnsShell::WindowContext::grTransactionLock_;
+#endif
 namespace facebook {
 namespace react {
 
-namespace RSkImageCacheManager{
-   
-namespace {
-
-  typedef better::map <string, sk_sp<SkImage>> ImageCacheMap;
-  ImageCacheMap imageCache;
-  #define CPU_MEM_ARR_INDEX 0
-  #define GPU_MEM_ARR_INDEX 1
-  std::mutex mutexLock;
-
-  inline sk_sp<SkImage> makeImageData(const char *path) {
-    sk_sp<SkData> data = SkData::MakeFromFileName(path);
-    if (!data) {
-      RNS_LOG_ERROR("Unable to make SkData for path :" << path);
-      return nullptr;
-    }
-    return( SkImage::MakeFromEncoded(data ));
-  }
-  
-  inline void getCacheUsage(size_t usageArr[]) {
-    int fOldCount{0};
-    usageArr[CPU_MEM_ARR_INDEX] = SkGraphics::GetResourceCacheTotalBytesUsed();
-  #ifdef RNS_SHELL_HAS_GPU_SUPPORT
-    GrDirectContext* gpuContext =RSkSurfaceWindow::getDirectContext();
-    if(gpuContext)
-      gpuContext->getResourceCacheUsage(&fOldCount, &usageArr[GPU_MEM_ARR_INDEX]);
-    else
-      usageArr[GPU_MEM_ARR_INDEX]=0;
-  #endif
-    RNS_LOG_DEBUG("CPU CACHE consumed bytes: "<<usageArr[CPU_MEM_ARR_INDEX]<< ", GPU CACHE consumed bytes: "<<usageArr[GPU_MEM_ARR_INDEX]);
-  }
+#define CPU_MEM_ARR_INDEX 0
+#define GPU_MEM_ARR_INDEX 1
+std::mutex imageCacheLock;
 
-  bool evictAsNeeded() {
-
-    int evictCount{0};
-    size_t usageArr[2]={0,0};
+RSkImageCacheManager& RSkImageCacheManager::getImageCacheManagerInstance() {
+  static RSkImageCacheManager imageCacheManagerInstance;
```

Comment:
Can we follow common singleton creation method like we have done in other components.